blog/2025-02-21/index.md

```diff
@@ -0,0 +1,86 @@
+
+![Flatpak logo](flatpak.png)
+
+From a technical perspective, Flatpak does not require elevated privileges to install apps, isolates apps from one another, and limits app access to the host environment. It makes deep use of existing Linux security technologies such as cgroups, namespaces, bind mounts, and seccomp as well as [Bubblewrap](https://github.com/containers/bubblewrap) for sandboxing.
+
+Flatpak apps are also built from a declarative manifest, which defines the exact sources to build from to enable as much reproducibility as possible.
+
+Due to Flatpak's sandboxing, apps don't have permission to access to many aspects of the host OS or user data they might need. To get that access, apps must request it using Portals or static permissions.
+
+### Portals & Static Permissions
+
+Most permissions can be requested and granted on demand via an API called [Portals](https://flatpak.github.io/xdg-desktop-portal/docs/). These permissions do not need to be given ahead of time, as desktop environments provide the mechanisms to give user consent and control over them e.g. by indicating their use, directly prompting the user before the permission is granted, and allowing revocation.
+
+![Illustration of portal, light](xdg-portal-light.png#gh-light-mode-only)
+![Illustration of a portal, dark](xdg-portal-dark.png#gh-dark-mode-only)
+
+Portals include APIs for handling auto-start and background activity; access to the camera, clipboard, documents, files, location, screen casting, screenshots, secrets like passwords, trash, and USB devices; setting global shortcuts; inhibiting suspend or shut down; capturing input; monitoring memory, network, or power profiles; sending notifications; printing; setting a wallpaper; and more. In each case, the user's desktop environment (like GNOME or KDE) manages if and how a user is notified or prompted for permissions—and if the permission is not granted, the app must handle it gracefully.
+
+Apps must also define any [static permissions](https://docs.flatpak.org/en/latest/sandbox-permissions.html) they need to function. These are clearly defined up front and include access to resources such as the network, Bluetooth, and audio devices. For regular file handling such as opening or saving, apps can use the File Chooser portal. In some cases apps may only make sense with permanent access to specific a folder, in which case a narrowly-scoped static permission (e.g. read-only access to the user's Music folder) may be used.
+
+Static permissions are designed to be as narrowly-scoped as possible and are unchanging per release of an app. They are not designed to be modified by an end user except in cases of development or debugging. Due to this, Flatpak always prefers apps to use Portals over static permissions whenever possible.
```

Comment:
maybe we should add that they can be "overriden" by the user using apps like Flatseal or KDE settings. This can be "security" feature too because technical users can remove permissions they don't need but at the cost of breaking the app.

"They are not designed to be modified by an end user" may give the impression that it's not possible to override them.

Reply:
Good point; I noticed that in the docs themselves as well. I can link there.